src/progly/run-papertests.cc

```diff
@@ -0,0 +1,312 @@
+    std::cout << "oidname      = " << oidname << std::endl ;
+    std::cout << "timestamp    = " << str << std::endl ;
+    std::cout << "--->exp : " << oidname << std::endl ;
+  }
+
+  if( write_type == 0 ) {
+    raw_whole_write( oidname, DATA_PATH, poolname, debug ) ;
+  }
+  else if( write_type == 1 ) {
+    raw_whole_write( oidname, DATA_PATH, poolname, debug ) ;
+  }
+  else {
+    std::cout << "unrecognized write_type '" << std::to_string( write_type ) << "'" << std::endl ;
+  }
+  std::string res = read_obj( oidname, optype, write_type, debug ) ;
+  res_file << res << "\n" ;
+  res_file.close() ;
+
+  return 0 ;
+}
+
+void raw_whole_write( std::string oid, std::string filename, std::string poolname, bool debug ) {
+
+  std::ifstream infile ;
+  infile.open( filename ) ;
+
+  std::string line ;
+  librados::bufferlist bl ;
+  while( std::getline( infile, line ) ) {
+    ::encode( line, bl ) ;
+  }
+
+  // save to ceph object
+  // connect to rados
+  librados::Rados cluster;
+  cluster.init(NULL);
+  cluster.conf_read_file(NULL);
+  int ret = cluster.connect();
+
+  // open pool
+  librados::IoCtx ioctx ;
+  ret = cluster.ioctx_create( poolname.c_str(), ioctx ) ;
+
+  // write bl_seq to ceph object
+  const char *obj_name = oid.c_str() ;
+  bufferlist::iterator p = bl.begin();
+  size_t num_bytes_written = p.get_remaining() ;
+  if( debug )
+    std::cout << "num_bytes_written : " << num_bytes_written << std::endl ;
+  ret = ioctx.write( obj_name, bl, num_bytes_written, 0 ) ;
+
+  ioctx.close() ;
+}
+
+void raw_iter_write( std::string oid, std::string filename, std::string poolname, bool debug ) {
+
+  std::ifstream infile ;
+  infile.open( filename ) ;
+
+  std::string line ;
+  librados::bufferlist bl ;
+  while( std::getline( infile, line ) ) {
+    //line.erase( std::remove( line.begin(), line.end(), '\n' ), line.end() ) ;
+    ::encode( line, bl ) ;
+  }
+
+  // save to ceph object
+  // connect to rados
+  librados::Rados cluster;
+  cluster.init(NULL);
+  cluster.conf_read_file(NULL);
+  int ret = cluster.connect();
+
+  // open pool
+  librados::IoCtx ioctx ;
+  ret = cluster.ioctx_create( poolname.c_str(), ioctx ) ;
+
+  // write bl_seq to ceph object
+  const char *obj_name = oid.c_str() ;
+  bufferlist::iterator p = bl.begin();
+  size_t num_bytes_written = p.get_remaining() ;
+  if( debug )
+    std::cout << "num_bytes_written : " << num_bytes_written << std::endl ;
+  ret = ioctx.write( obj_name, bl, num_bytes_written, 0 ) ;
+
+  ioctx.close() ;
+}
+
+void do_sum( librados::bufferlist bl_seq, int write_type, bool debug ) {
+
+  std::vector< uint64_t > these_ints ;
+
+  // raw
+  if( write_type == 0 ) {
+    ceph::bufferlist::iterator it_bl_seq = bl_seq.begin() ;
+    ceph::bufferlist bl ;
+    ::decode( bl, it_bl_seq ) ;
+    const char* raw = bl.c_str() ;
+    int count = 0 ;
+    std::string tmp = "" ;
+
+    for( int i = 0; i < std::strlen( raw ); i++ ) {
+      tmp = tmp + raw[i] ;
```

Comment:
is this a single long string with single digit ints i guess?

Reply:
yup. tmp collects characters and will be flushed as a complete 6-digit int after hitting 6 characters.